src/components/Explore/Map.tsx

```diff
@@ -256,43 +251,35 @@ const ExploreMap = (): JSX.Element => {
         // Update visible stations when the given dependencies change
         const map = mapRef.current;
         if (map && isMapLoaded) {
-            if (
-                filteredStations.length ||
-                filteredFAOAreas.length ||
-                filteredSpecies.length ||
-                filterDates[0] !== null ||
-                filterDates[1] !== null
-            ) {
-                searchStations(
-                    {
-                        stationNames: filteredStations,
-                        faoAreas: filteredFAOAreas,
-                        species: filteredSpecies,
-                        dates: filterDates
-                    },
-                    (stations) => {
-                        map.setFilter('stations', ['in', 'name', ...stations.map((station) => station.name)]);
-                        // Move the map to filtered stations
-                        if (stations.length === 1) {
-                            map.flyTo({
-                                center: stations[0].coordinates,
-                                zoom: 6
-                            });
-                        } else {
-                            const bounds = getFeatureBounds(
-                                stations.map(({ coordinates }) => coordinates) as LineCoordinates
-                            );
-                            if (!bounds.isEmpty()) {
-                                map.fitBounds(bounds, { padding: 50 });
-                            }
-                        }
-
-                        if (selectedStation && !stations.find(({ name }) => name === selectedStation.name)) {
-                            dataActionDispatcher({ type: 'updateSelectedStation', station: null });
-                        }
-                        filterActionDispatcher({ type: 'updateFilterCount', count: stations.length });
+            if (filteredStations) {
+                map.setFilter('stations', ['in', 'name', ...filteredStations.map((station) => station.name)]);
+                // Move the map to filtered stations
+                if (filteredStations.length === 1) {
+                    map.flyTo({
+                        center: filteredStations[0].coordinates,
+                        zoom: 6
+                    });
+                } else {
+                    const bounds = getFeatureBounds(
+                        filteredStations.map(({ coordinates }) => coordinates) as LineCoordinates
+                    );
+                    if (!bounds.isEmpty()) {
+                        // map.fitBounds(bounds, {
```

Comment:
The filter fly effect doesn't work because of this. What is the issue you are facing?

Reply:
Also, when you follow these steps, the map stops flying to the selected station even for single selection.
- Select 1 species, the map will fly to the station as expected
- Select another species, the map will now zoom out and max out the bounds
- Now clear and select 1 species, this time the map won't fly to that single point.